README.md

```diff
@@ -1,119 +1,441 @@
+
+### Public Properties ###
+
+The Message class doesn't contain any getters, however there are a couple of public properties that may want to be accessed and used. See *onAck* and *onFail* method examples for details.
+
+| Property Name | Data&nbsp;Type | Value | 
+| --- | --- | --- |
+| *payload* | Table | A table that is sent between partners when the send method is called. See [Payload Table](#payload-table) below for table details. |
+| *metadata* | Any serializable type | Optional application defined data containing additional message properties. This data **WILL NOT** be sent to the partner, but can be accessed when the *onAck* or *onFail* handler are triggered. | 
+
+#### Payload Table ####
+
+| Key | Data&nbsp;Type | Description |
+| --- | --- | --- | 
+| *id* | Integer | Message identifier, an auto-incrementing integer |
+| *name* | String | Name of message |
+| *data* | Any serializable type | Message data |
+
+## ReplayMessenger Usage ##
+
+**ReplayMessenger** extends the **Messenger** class adding features for resending and persisting messages. **ReplayMessenger** also has dependencies on [**SPIFlashLogger**](https://github.com/electricimp/SpiFlashLogger) and [**ConnectionManager**](https://github.com/electricimp/ConnectionManager) libraries. 
+
+### Constructor: ReplayMessenger(*spiFlashLogger, cm[, options]*) ###
+
+Calling the **ReplayMessenger** constructor creates a new **ReplayMessenger** instance. 
+
+#### Parameters ####
+
+| Parameter | Type | Required? | Description |
+| --- | --- | --- | --- |
+| *spiFlashLogger* | Instance of spiFlashLogger | Yes | Instance of [spiFlashLogger library](https://github.com/electricimp/SpiFlashLogger) which will be used to store messages. Must include library v2.2.0 or above. |
+| *connectionManager* | Instance of ConnectionManager | Yes | Instance of [ConnectionManager](https://github.com/electricimp/ConnectionManager) which will be used to check the connection state. Must include library v3.1.0 or above. |
+| *options* | Table | No | An optional table with settings that override default behaviors. See [Options Table](#options-table) below for details. Default: `{}` |
+
+#### ReplayMessenger Options Table ####
+
+| Key | Data&nbsp;Type | Description |
+| ---- | --- | --- |
+| *debug* | Boolean | The flag that enables debug library mode, which turns on extended logging. Default: `false` |
+| *ackTimeout* | Integer | Changes the default timeout required before a message is considered failed (to be acknowledged). Default: 10s |
+| *maxRate* | Integer | Maximum message send rate, which defines the maximum number of messages the library allows to send per second. If the application exceeds the limit, the *onFail* callback is triggered. Default: 10 messages per second<br />**Note** please don’t change the value unless absolutely necessary |
+| *firstMsgId* | Integer | Initial value for the auto-incrementing message ID. Default: 0 |

```

Comment:
I would add here that this ID should be picked in such a way that it will not interfere with the messages currently persisted on the flash. The general rule here is to pick an ID which is greater than the IDs of the currently persisted messages. If the firstMsgId parameter is omitted, the ID will be picked automatically by checking all currently persisted messages.